app/src/components/Register.tsx

```diff
@@ -197,12 +203,42 @@ class Register extends React.Component<RegisterProps, RegisterState> {
     this.setState(stateUpdate);
   };
 
+  artistChangeHandler = (event: InputChangeEvent): void => {
+    if (!this.state.artists) {
+      return;
+    }
+
+    const value = event.target.value;
+
+    let artist;
+
+    for (const a of this.state.artists) {
+      if (a.name === value) {
+        artist = a;
+        break;
+      }
+    }
+
+    if (!artist) {
+      return;
+    }
+
+    const stateUpdate = { fields: this.state.fields as Pick<RegisterFormFields, keyof RegisterFormFields> };
+    stateUpdate.fields.artistName = artist.name;
+    stateUpdate.fields.artistNationality = artist.nationality;
+    stateUpdate.fields.artistBirthYear = artist.birthYear;
```

Comment:
Is this function meant to call this.setState(stateUpdate) at the end or does the state get updated elsewhere?

Reply:
Hmmm interesting one. Now you mention it I would probably say yes as the previous function uses it. However on testing it appears to work so not quiet sure

Reply:
Added it to be safe